drivers/clock_control/clock_stm32_ll_u5.c

```diff
@@ -303,82 +303,19 @@ void config_src_sysclk_pll(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
 	LL_RCC_PLL1_SetQ(STM32_PLL_Q_DIVISOR);
 #endif /* STM32_PLL_Q_DIVISOR */
 
-#if STM32_PLL_SRC_MSIS
-
-	/*
-	 * For now, an issue detected in function LL_PLL_ConfigSystemClock_MSI
-	 * doesn't allow it's use in current Cube package version (1.0.0).
-	 * So we're using step by step configuration, with fixed flash latency
-	 * setting.
-	 * This has been tested using max supported freq (160MHz), but could
-	 * have limitations in lower speed settings.
-	 */
-
-	LL_FLASH_SetLatency(LL_FLASH_LATENCY_4);
-	while (LL_FLASH_GetLatency() != LL_FLASH_LATENCY_4) {
-	}
-
-	set_up_clk_msis();
-
-	LL_RCC_PLL1_ConfigDomain_SYS(LL_RCC_PLL1SOURCE_MSIS,
-				     STM32_PLL_M_DIVISOR,
-				     STM32_PLL_N_MULTIPLIER,
-				     STM32_PLL_R_DIVISOR);
-
 	set_regu_voltage(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
 
-	if (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC >= 55) {
-		/*
-		 * Set EPOD prescaler based on PLL1 input freq (MSI/PLLM)
-		 * Booster clock frequency should be between 4 and 16MHz
-		 * This is done in following steps:
-		 * Read MSI Frequency
-		 * Didvide PLL1 input freq (MSI/PLLM) by the targeted freq (8MHz)
-		 * Make sure value is not higher than 16
-		 * Shift in the register space (/2)
-		 */
-		int tmp = __LL_RCC_CALC_MSIS_FREQ(LL_RCC_MSIRANGESEL_RUN,
-				STM32_MSIS_RANGE << RCC_ICSCR1_MSISRANGE_Pos);
-		tmp = MIN(tmp / STM32_PLL_M_DIVISOR / 8000000, 16);
-		tmp = tmp / 2;
-		LL_RCC_SetPll1EPodPrescaler(tmp << RCC_PLL1CFGR_PLL1MBOOST_Pos);
-
-		LL_PWR_EnableEPODBooster();
-		while (LL_PWR_IsActiveFlag_BOOST() == 0) {
-		}
-	}
-
-	LL_RCC_PLL1_EnableDomain_SYS();
-	LL_RCC_PLL1_SetVCOInputRange(LL_RCC_PLLINPUTRANGE_4_8);
-
-	LL_RCC_PLL1_Enable();
-
-	/* Wait till PLL is ready */
-	while (LL_RCC_PLL1_IsReady() != 1) {
-	}
-
-	/* Intermediate AHB prescaler 2 for target frequency clock > 80 MHz */
-	LL_RCC_SetAHBPrescaler(LL_RCC_SYSCLK_DIV_2);
-	LL_RCC_SetSysClkSource(LL_RCC_SYS_CLKSOURCE_PLL1);
-
-	/* Wait till System clock is ready */
-	while (LL_RCC_GetSysClkSource() != LL_RCC_SYS_CLKSOURCE_STATUS_PLL1) {
-	}
-
-	LL_RCC_SetAHBPrescaler(s_ClkInitStruct.AHBCLKDivider);
-	LL_RCC_SetAPB1Prescaler(s_ClkInitStruct.APB1CLKDivider);
-	LL_RCC_SetAPB2Prescaler(s_ClkInitStruct.APB2CLKDivider);
-	LL_RCC_SetAPB3Prescaler(s_ClkInitStruct.APB3CLKDivider);
+#if STM32_PLL_SRC_MSIS
+	set_up_clk_msis();
 
-	LL_SetSystemCoreClock(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
+	/* Switch to PLL with MSI as clock source */
```

Comment:
It looks to me like we loose VCO input configuration and EPOD configuration. Is it ?

Reply:
Indeed, maybe I was a bit fast in removing all this. This being said, it works in all config that I have been able to test.
One thing is that none of these is specific to MSI, so it should actually be extented to other PLL sources.
I'll have a look

Reply:
@ABOSTM Looks like implementing this for all PLL will be easier on the curated U5 driver version proposed in https://github.com/zephyrproject-rtos/zephyr/pull/41585. So I propose to log this in an issue and implemented on top of it.